packages/js/components/src/form/form.tsx

```diff
@@ -0,0 +1,250 @@
+/**
+ * External dependencies
+ */
+import {
+	cloneElement,
+	useState,
+	createElement,
+	useCallback,
+	useEffect,
+} from '@wordpress/element';
+import deprecated from '@wordpress/deprecated';
+import { ChangeEvent, PropsWithChildren } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { FormContext } from './form-context';
+
+type FormProps< Values > = {
+	/**
+	 * Object of all initial errors to store in state.
+	 */
+	errors: Record< keyof Values, string >;
+	/**
+	 * Object key:value pair list of all initial field values.
+	 */
+	initialValues: Values;
+	/**
+	 * This prop helps determine whether or not a field has received focus
+	 */
+	touched: Record< keyof Values, boolean >;
+	/**
+	 * Function to call when a form is submitted with valid fields.
+	 *
+	 * @deprecated
+	 */
+	onSubmitCallback?: ( values: Values ) => void;
+	/**
+	 * Function to call when a form is submitted with valid fields.
+	 */
+	onSubmit?: ( values: Values ) => void;
+	/**
+	 * Function to call when a value changes in the form.
+	 *
+	 * @deprecated
+	 */
+	onChangeCallback?: () => void;
+	/**
+	 * Function to call when a value changes in the form.
+	 */
+	onChange?: (
+		value: { name: string; value: unknown },
```

Comment:
Why is it `value: unknown` when `Values` is `Record< string, string >`?

Reply:
Good point actually, I might of missed this ( to many occurrences of close to the same types) I replace the `unknown` with `any` and replaced `Record< string, string >` with `Record< string, any >` to more accurately reflect a Form object.